website/blog/2023-09-06-optimize-try-cast.mdx

```diff
@@ -0,0 +1,114 @@
+---
+slug: optimize-try_cast.
+title: "Learnings from optimizing try_cast"
+authors: [lsakka]
+tags: [tech-blog, performance]
+---
+
+One of the queries shadowed internally at Meta was much slower in Velox compared to presto(2 days vs 4.5h). Initial investigation identified that the overhead is related to casting empty strings inside a try_cast.
+
+In this blogpost I summarize my learnings from investigating and optimizing try_cast.
+
+--------------------------------------------------------------------------------------------
+
+### Start and end results
+
+Initial benchmark:
+```
+try_cast(empty_string_col as int)                     4.88s   204.83m
```

Comment:
Were all the values in empty_string_col matching strings of integer/numeric value ? This expression only seems reasonable in that context. If yes, then what is the semantics of empty string in such an input ? Shouldn't we expect null values instead ?

Reply:
all the values in empty_string_col are ''

Reply:
result is null.

Reply:
try_cast('' as int) except that '' is in flat vector